src/app/courses/step-view-courses/courses-step-view.component.ts

```diff
@@ -177,4 +178,36 @@ export class CoursesStepViewComponent implements OnInit, OnDestroy {
     });
   }
 
+  surveyExists(): boolean {
+    const exists = this.stepDetail && this.stepDetail.survey &&
```

Comment:
A function where you assign a variable and then return that variable should simply be refactored to return the value.